### DISCUSSION TITLE
Unable to inject Service

### DISCUSSION THREAD

**Discussion by DeckardKain** [betalgo/openai#46] [Q&A] [answered]
Most likely a newb question but when I go to add this service to my .net 6 blazor app in the program.cs 

builder.Services.AddOpenAIService();

I get:

Unable to resolve service for type 'OpenAI.GPT3.Managers.OpenAIService' while attempting to activate 'LegendariumUI.Services.AIService'.) 

I tried copying what you had in your program.cs file however, this line doesnt work for me

IConfiguration configuration = builder.Build();

I appreciate any assistance here.

Thank you!

**Comment by DeckardKain** [comment, marked as answer]
Ok, I was able to resolve my problem by dependency injecting the interface into my AIService class instead of the class itself. Doh!

Though, when I go to get a picture, the program seems to be saving my User field and Im getting errors saying Code = invalid_organization even though I have put a different value in there.

**Reply by kayhantolga** [reply]
What is your version number?

**Reply by DeckardKain** [reply]
6.6.1

**Comment by awonuga123** [comment]
I'm lost pls, what's going on?

203900

**Reply by kayhantolga** [reply]
There were some spam replies, please ignore them, I reported them already.
Which version of Betalgo.OpenAI.GPT3 you are using?

**Reply by DeckardKain** [reply]
I am using 6.6.1

**Reply by kayhantolga** [reply]
there was an issue with the organization in the previous version but you are already using the latest version, so it should be fine. Could you share your setting file (with hiding values)? 
And what is the error Message?

**Reply by DeckardKain** [reply]
OK, i have gotten it all to work. Now, how to get unlimited calls!

Thank you kayhantolga!

**Reply by ADefWebserver** [reply]
I posed a Blazor .Net 7 example using this library at [Calling OpenAI GPT-3 From Microsoft Blazor](https://blazorhelpwebsite.com/ViewBlogPost/65)
